WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift

```diff
@@ -0,0 +1,295 @@
+import SwiftUI
+import DesignSystem
+
+struct CommentModerationView: View {
+    @StateObject private var viewModel: CommentModerationViewModel
+
+    init(viewModel: CommentModerationViewModel) {
+        _viewModel = StateObject(wrappedValue: viewModel)
+    }
+
+    var body: some View {
+        VStack(spacing: .DS.Padding.double) {
+            Divider()
+                .foregroundStyle(Color.DS.Background.secondary)
+            VStack(spacing: .DS.Padding.double) {
+                switch viewModel.state {
+                case .pending:
+                    Pending(viewModel: viewModel)
+                case .approved(let liked):
+                    Approved(viewModel: viewModel, liked: liked)
+                case .trash, .spam:
+                    TrashSpam(viewModel: viewModel)
+                }
+            }
+            .padding(.horizontal, .DS.Padding.double)
+        }
+        .animation(.smooth, value: viewModel.state)
+    }
+}
+
+// MARK: - Subviews
+
+private struct Container<T: View, V: View>: View {
+
+    let title: String
+    let icon: V
+    let content: T
+
+    private var transition: AnyTransition = .opacity
+
+    private let animationDuration = 0.25
+
+    private let insertionAnimationDelay = 0.8
+
+    private var insertionAnimation: Animation {
+        return removalAnimation.delay(insertionAnimationDelay)
```

Comment:
Hey @alpavanoglu There was a small regression when I pushed this commit https://github.com/wordpress-mobile/WordPress-iOS/pull/23127/commits/fc2c17644f4e5ac1406b0ccb6204b1f5f1a92aa4

```suggestion
        return removalAnimation.delay(animationDuration * insertionAnimationDelay)
```